pkg/kapp/resources/identified_resources_pods.go

```diff
@@ -35,31 +37,33 @@ func (w UniquePodWatcher) Watch(podsToWatchCh chan corev1.Pod, cancelCh chan str
 		// Watch Pods in all namespaces first and fallback to the
 		// fallbackAllowedNamespaces if lack of permission
 		namespace := ""
+		var index int
+		var forbiddenNamespaces []string
 		for {
 			podWatcher := NewPodWatcher(
 				w.coreClient.CoreV1().Pods(namespace),
 				metav1.ListOptions{LabelSelector: w.labelSelector.String()},
 			)
-
 			err := podWatcher.Watch(nonUniquePodsToWatchCh, cancelCh)
-			if err == nil {
-				break
-			}
-			if errors.IsForbidden(err) && namespace == "" {
-				// The '-n' flag or default state namespace can specify only 1 namespace, so there
-				// should be at most 1 item in fallbackAllowedNamespaces
-				if len(w.fallbackAllowedNamespaces) > 0 {
-					namespace = w.fallbackAllowedNamespaces[0]
-					if namespace == "" {
+			if err != nil {
+				if namespace != "" {
+					if !errors.IsForbidden(err) {
+						fmt.Printf("Pod watching error: %s\n", err) // TODO
 						break
 					}
+					forbiddenNamespaces = append(forbiddenNamespaces, fmt.Sprintf(`"%s"`, namespace))
 				}
-			} else {
-				fmt.Printf("Pod watching error: %s\n", err) // TODO
+			}
+
+			if index >= len(w.fallbackAllowedNamespaces) {
```

Comment:
Why is this the only breaking condition on success, do we not wanna stop watching the moment we can watch all namespaces?

Reply:
Looks like we only stop if
1. There is an error which is not a forbidden error
2. We have watched *all* the fallback allowed namespaces and tried watching all namespaces as well.

Reply:
> do we not wanna stop watching the moment we can watch all namespaces?

Yes, good catch. Made the changes.